### PR TITLE
fix(community): For IBM implementation rename variables, remove defaults, fix tests and minor docs fixes

### PR DESCRIPTION
**Pull request by FilipZmijewski**
Following PR provides following fixes:

- rename variables to camelCase
- remove default values for model
- Update and remove some tests
- update urls in documentation, shrink outputs

### CODE REVIEW

**Comment by FilipZmijewski** [comment]
@jacoblee93 Could you please have a look at it or assign someone to this PR? Thank you

**Comment by jacoblee93** [comment]
Happy to merge since this is a newer integration but noting that this is breaking

**Comment by jacoblee93** [comment]
Looks like you need to update your notebooks, they are failing our CI check:

```
tmp/ibm.mts (40,5): Object literal may only specify known properties, but 'max_new_tokens' does not exist in type 'WatsonxInputLLM'. Did you mean to write 'maxNewTokens'?
file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/check_notebook_type_errors.js:101
            const err = new Error("Found type errors in new notebook.");
                        ^

Error: Found type errors in new notebook.
    at checkNotebookTypeErrors (file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/check_notebook_type_errors.js:101:[25](https://github.com/langchain-ai/langchainjs/actions/runs/11612088879/job/32550548449?pr=7129#step:8:26))
    at main (file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/index.js:4:24)
    at file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/index.js:7:10
    at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:337:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:106:12) {
  details: "tmp/ibm.mts (40,5): Object literal may only specify known properties, but 'max_new_tokens' does not exist in type 'WatsonxInputLLM'. Did you mean to write 'maxNewTokens'?"
}

Node.js v18.20.4
tmp/ibm.mts (9,3): Object literal may only specify known properties, and 'idOrName' does not exist in type 'WatsonxInputEmbeddings & WatsonxAuth'.
file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/check_notebook_type_errors.js:101
            const err = new Error("Found type errors in new notebook.");
                        ^

Error: Found type errors in new notebook.
    at checkNotebookTypeErrors (file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/check_notebook_type_errors.js:101:25)
    at main (file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/index.js:4:24)
    at file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/index.js:7:10
    at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:[33](https://github.com/langchain-ai/langchainjs/actions/runs/11612088879/job/32550548449?pr=7129#step:8:34)7:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:106:12) {
  details: "tmp/ibm.mts (9,3): Object literal may only specify known properties, and 'idOrName' does not exist in type 'WatsonxInputEmbeddings & WatsonxAuth'."
}
```

File: libs/langchain-community/src/chat_models/ibm.ts [hunk 1]
```diff
@@ -81,30 +79,8 @@ export interface WatsonxDeltaStream {
...
-      | "maxTokens"
-      | "presencePenalty"
-      | "responseFormat"
-      | "timeLimit"
-      | "modelId"
-    >
-  > {
+  extends Partial<Omit<TextChatParams, "modelId">> {
   maxRetries?: number;
-  tool_choice?: TextChatToolChoiceTool;
```
[the remark this answers is not in the document]
Comment (FilipZmijewski) [reply libs/langchain-community/src/chat_models/ibm.ts:99]: Indeed variable's names changes are breaking, but this needs to be done

**Comment by FilipZmijewski** [comment]
I fixed the failing notebook

**Comment by jacoblee93** [comment]
Now failing lint and:

```
tmp/ibm.mts (39,3): Type '{ maxNewTokens: number; }' is missing the following properties from type 'WatsonxInputLLM': model, serviceUrl, version
file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/check_notebook_type_errors.js:101
            const err = new Error("Found type errors in new notebook.");
                        ^

Error: Found type errors in new notebook.
    at checkNotebookTypeErrors (file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/check_notebook_type_errors.js:101:[25](https://github.com/langchain-ai/langchainjs/actions/runs/11701931027/job/32611842791?pr=7129#step:8:26))
    at main (file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/index.js:4:24)
    at file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/index.js:7:10
    at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:337:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:106:12) {
  details: "tmp/ibm.mts (39,3): Type '{ maxNewTokens: number; }' is missing the following properties from type 'WatsonxInputLLM': model, serviceUrl, version"
}

Node.js v18.20.4
tmp/ibm.mts (9,3): Object literal may only specify known properties, and 'idOrName' does not exist in type 'WatsonxInputEmbeddings & WatsonxAuth'.
file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/check_notebook_type_errors.js:101
            const err = new Error("Found type errors in new notebook.");
                        ^

Error: Found type errors in new notebook.
    at checkNotebookTypeErrors (file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/check_notebook_type_errors.js:101:25)
    at main (file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/index.js:4:24)
    at file:///home/runner/work/langchainjs/langchainjs/libs/langchain-scripts/dist/notebooks/index.js:7:10
    at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:[33](https://github.com/langchain-ai/langchainjs/actions/runs/11701931027/job/32611842791?pr=7129#step:8:34)7:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:106:12) {
  details: "tmp/ibm.mts (9,3): Object literal may only specify known properties, and 'idOrName' does not exist in type 'WatsonxInputEmbeddings & WatsonxAuth'."
}
```

**Comment by FilipZmijewski** [comment]
I am just fixing the notebooks but lint doesn't seem to be a problem with our implementation:
```
/home/runner/work/langchainjs/langchainjs/libs/langchain-xai/src/tests/chat_models_structured_output.int.test.ts
Error:   2:1  error  'zod-to-json-schema' should be listed in the project's dependencies. Run 'npm i -S zod-to-json-schema' to add it  import/no-extraneous-dependencies 
```

**Comment by jacoblee93** [comment]
CC @bracesproul above

**Comment by FilipZmijewski** [comment]
> CC @bracesproul above

any news why the lint would be failing?

**Comment by jacoblee93** [comment]
Have been told it merging main should fix it, have just done os!

**Comment by jacoblee93** [comment]
Thanks!

**Comment by FilipZmijewski** [comment]
Thank you!

**Pull request merged**